docs/concepts/executor/docarray-v2.md

```diff
@@ -0,0 +1,121 @@
+(docarray-v2)=
+
+
+# (Beta) Docarray-V2
+
+Jina provide an early support for [DocArray-v2](https://github.com/docarray/docarray/commits/feat-rewrite-v2) which
+is a rewrite of DocArray.
+
+```{warning} Beta support
+DocArray v2 is  still in alpha and its support in Jina is still an experimental feature, and the API is subject to 
+change.
+```
+
+## DocArray v2 schema
+
+At the heart of DocArray v2 is a new schema that is more flexible and expressive than the original DocArray schema.
+
+You can refer to the [DocArray v2 readme](https://github.com/docarray/docarray/tree/feat-rewrite-v2) for more details.
+
+
+On Jina side this has actually quite a big impact on the spirit of how we are building Executor. Indeed, with docarray v1
+the version that is currently used in Jina, the `Document` as a fixed schema and the Executor perform in place operation
+on directly on it. With DocArray v2 things change slightly. Each executor will need to define its own input schema
+and output schema. Of course, we still provided predefined schema. We believe this will allow Executor to be more
+expressive and flexible.
```

Comment:
```suggestion
the version that is currently used in Jina, the `Document` as a fixed schema and the Executor performs in place operations
on directly on it. With DocArray v2 things change slightly. Each executor will need to define its own input schema
and output schema. Of course, we still provided predefined schema. We believe this will allow Executor to be more
expressive and flexible.
```